src/filters/uniform.rs

```diff
@@ -72,39 +68,46 @@ pub fn uniform_filter1d<S, A, D>(
 where
     S: Data<Elem = A>,
     A: Float + FromPrimitive + 'static,
-    for<'a> &'a [A]: SymmetryStateCheck,
     D: Dimension,
 {
-    let weights = weights(size);
     let mut output = array_like(&data, data.dim(), A::zero());
-    inner_correlate1d(data, &weights, axis, mode, 0, &mut output);
+    inner_uniform1d(data, size, axis, mode, &mut output);
     output
 }
 
-fn weights<A>(size: usize) -> Vec<A>
-where
-    A: Float + FromPrimitive + 'static,
+pub(crate) fn inner_uniform1d<S, A, D>(
+    data: &ArrayBase<S, D>,
+    size: usize,
+    axis: Axis,
+    mode: BorderMode<A>,
+    output: &mut Array<A, D>,
+) where
+    S: Data<Elem = A>,
+    A: Copy + Num + FromPrimitive + PartialOrd,
+    D: Dimension,
 {
-    if size == 0 {
-        panic!("Size is zero");
-    }
-    [A::one() / A::from_usize(size).unwrap()].repeat(size)
-}
+    let size1 = size / 2;
+    let size2 = size - size1 - 1;
+    let size_as_a = A::from_usize(size).unwrap();
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use approx::assert_relative_eq;
+    let mode = mode.to_pad_mode();
+    let n = data.len_of(axis);
+    let pad = vec![[size1, size2]];
+    let mut buffer = Array1::from_elem(n + size - 1, mode.init());
 
-    #[test]
-    fn test_weights() {
-        assert_relative_eq!(weights(5).as_slice(), &[0.2, 0.2, 0.2, 0.2, 0.2][..], epsilon = 1e-7);
-        assert_relative_eq!(weights(1).as_slice(), &[1.0][..], epsilon = 1e-7);
-    }
+    Zip::from(data.lanes(axis)).and(output.lanes_mut(axis)).for_each(|input, mut o| {
+        pad_to(&input, &pad, mode, &mut buffer);
+        let buffer = buffer.as_slice_memory_order().unwrap();
+        let mut accumulator = buffer.iter().take(size - 1).fold(A::zero(), |acc, elem| acc + *elem);
```

Comment:
Can you write this instead?

    let mut accumulator: A = buffer[..size].iter().sum();

or remove the `as_slice_memory_order` line and

    let mut accumulator = buffer.slice(s![..size]).sum()